samples/mqtt/sender/main.go

```diff
@@ -62,7 +61,7 @@ func main() {
 			log.Printf("failed to set data: %v", err)
 		}
 		if result := c.Send(
-			cecontext.WithTopic(ctx, "test-topic"),
+			cemqtt.WithProtocolContext(ctx, cemqtt.NewProtocolContext("test-topic", 0, false)),
```

Comment:
I can see this list of parameters getting quiet out of hand vs using functional options or a config struct instead.

Reply:
To keep the code short, such a `New()` constructor is used here. There is indeed some confusion when the parameter is not identified.

But cause we define several pointers in the `ProtocolContext`, using a config struct will significantly increase the amount of code(typing). 
Using functional options to pass multiple parameters will also increase the number of lines of configuration code.

I'm sure which one is better, and which of them do you suggest?

Reply:
We already use functional options and short (w/ default) constructors in other places so that IMHO would remove the need for `ProtocolContext` completely. WDYT?

Reply:
The short constructor is used to initialize the `ProtocolContext`. But I think you have given a good option that we can make use of the [struct option](https://github.com/eclipse/paho.golang/blob/86185b4d2a6c372afe12535368e2dcf805278911/paho/cp_publish.go#L12-L32) directly instead of `ProtocolContext`